data/src/main/java/com/microsoft/azure/kusto/data/auth/CloudInfo.java

```diff
@@ -0,0 +1,132 @@
+package com.microsoft.azure.kusto.data.auth;
+
+import com.microsoft.azure.kusto.data.exceptions.DataClientException;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+class CloudInfo {
+    public static final String METADATA_ENDPOINT = "v1/rest/auth/metadata";
+    public static final String DEFAULT_KUSTO_CLIENT_APP_ID = "db662dc1-0cfe-4e1c-a843-19a68e65be58";
+    public static final boolean DEFAULT_LOGIN_MFA_REQUIRED = false;
+    public static final String DEFAULT_PUBLIC_LOGIN_URL = "https://login.microsoftonline.com";
+    public static final String DEFAULT_REDIRECT_URI = "https://microsoft/kustoclient";
+    public static final String DEFAULT_KUSTO_SERVICE_RESOURCE_ID = "https://kusto.kusto.windows.net";
+    public static final String DEFAULT_FIRST_PARTY_AUTHORITY_URL = "https://login.microsoftonline.com/f8cdef31-a31e-4b4a-93e4-5f571e91255a";
+    public static final CloudInfo DEFAULT_CLOUD = new CloudInfo(
+            DEFAULT_LOGIN_MFA_REQUIRED,
+            DEFAULT_PUBLIC_LOGIN_URL,
+            DEFAULT_KUSTO_CLIENT_APP_ID,
+            DEFAULT_REDIRECT_URI,
+            DEFAULT_KUSTO_SERVICE_RESOURCE_ID,
+            DEFAULT_FIRST_PARTY_AUTHORITY_URL
+    );
+
+    private static final Map<String, CloudInfo> cache = new HashMap<>();
+
+    private final boolean loginMfaRequired;
+    private final String loginEndpoint;
+    private final String kustoClientAppId;
+    private final String kustoClientRedirectUri;
+    private final String kustoServiceResourceId;
+    private final String firstPartyAuthorityUrl;
+
+    public CloudInfo(boolean loginMfaRequired, String loginEndpoint, String kustoClientAppId, String kustoClientRedirectUri, String kustoServiceResourceId, String firstPartyAuthorityUrl) {
```

Comment:
Any reason for this to be public?

Reply:
Well it's mainly for the `manuallyAddToCache` function.
I needed to create it for the tests, and I thought there is no harm in making in public.
I can make both private/protected/package-private if you prefer

Reply:
Tests are important, so keep it :)
Just wanted to make sure